clang/lib/Sema/SemaChecking.cpp

```diff
@@ -7795,6 +7795,20 @@ bool Sema::CheckIntelSYCLPtrAnnotationBuiltinFunctionCall(unsigned BuiltinID,
   return false;
 }
 
+static llvm::APSInt getSYCLAllocaDefaultSize(const ASTContext &Ctx,
+                                             const VarDecl *VD) {
+  assert(VD && "Expecting valid declaration");
+  APValue *SpecializationId = VD->evaluateValue();
+  assert(SpecializationId && "Expecting value");
+  assert(SpecializationId->getKind() == APValue::ValueKind::Struct &&
+         "Expecting specialization_id struct");
+  assert(SpecializationId->getStructNumFields() == 1 &&
+         "Expecting single field for default value");
+  APValue Default = SpecializationId->getStructField(0);
+  assert(Default.isInt() && "Expecting default integer value");
```

Comment:
Update assert message as well.
 // Assert that the default value is of integer type.
  assert(Default.isInt() && "Expected the default value to be an integer");